msg-socket/src/req/socket.rs

```diff
@@ -61,18 +64,39 @@ where
         response_rx.await.map_err(|_| ReqError::SocketClosed)?
     }
 
-    /// Connects to the target with the default options.
+    /// Connects to the target with the default options. WARN: this will block until the connection can be established.
     pub async fn connect(&mut self, endpoint: SocketAddr) -> Result<(), ReqError> {
         // Initialize communication channels
         let (to_driver, from_socket) = mpsc::channel(DEFAULT_BUFFER_SIZE);
 
         tracing::info!("Connecting to {}", endpoint);
 
-        let stream = self
-            .transport
-            .connect(endpoint)
-            .await
-            .map_err(|e| ReqError::Transport(Box::new(e)))?;
+        let mut backoff = ExponentialBackoff::new(Duration::from_millis(20), 256);
```

Comment:
I suggest we cap the max retries to around 15?
that should be already enough where the last try will have 10min of backoff

Reply:
Yeah that's fair, 256 retries is infinity